test/functional/test_framework/test_node.py

```diff
@@ -97,14 +97,16 @@ def __init__(self, i, datadir, *, chain, rpchost, timewait, bitcoind, bitcoin_cl
             "-debugexclude=leveldb",
             "-uacomment=testnode%d" % i,
         ]
-        if use_valgrind:
+        if self.is_valgrind_present() and use_valgrind:
+            exit_on_first_error = self.should_exit_valgrind_on_first_error()
             default_suppressions_file = os.path.join(
                 os.path.dirname(os.path.realpath(__file__)),
                 "..", "..", "..", "contrib", "valgrind.supp")
             suppressions_file = os.getenv("VALGRIND_SUPPRESSIONS_FILE",
                                           default_suppressions_file)
             self.args = ["valgrind", "--suppressions={}".format(suppressions_file),
-                         "--gen-suppressions=all", "--exit-on-first-error=yes",
+                         "--gen-suppressions=all",
+                         "--exit-on-first-error=yes" if exit_on_first_error else "--exit-on-first-error=no",
```

Comment:
I haven't tested this, but I expect `--exit-on-first-error=no` to throw an error on versions that don't have this option.

Reply:
Will test it on my macOS Catalina, because valgrind can't be installed there. :)

Reply:
You mean the previous comment? This one is easiest to test on e.g. a fresh Ubuntu 18.04 machine, which has valgrind 3.13. You can then compile it from source and install 3.15 to see the difference.

Reply:
Yeah, I have one cheap VPS running, with Ubuntu 18.04. Will follow your recommendation. Thanks!